Added new Endpoints

dtibbe opened the pull request:
Hi,

I've added some new endpoints to the SparkClient.
Some existing endpoints were updated according to current results as they've been extended.

dtibbe committed:
* Several API Calls updated
* Added support for new API Calls like Devices
* Added support for Call Control API

dtibbe committed:
* Update to People model
* Added Recordings Endpoint

dtibbe committed:
* Added Recordings Endpoint

dtibbe committed:
* Added Events Endpoint
* Added AdminAudit Endpoint

dtibbe committed:
* Added Hybrid Endpoint

dtibbe committed:
Documentation updates for domain models

dtibbe committed:
Model documentation updates

dtibbe committed:
* Model documentation update
* Added support for xAPI endpoint
* Added some convinient methods
* Renamed PostCall* methods to Call*

darrenparkinson commented:
Awesome updates. Just hadn’t got around to adding the updates so that’s really appreciated.

dtibbe committed:
* fixed code errors
* Code cleanup
* Introduced common abstract superclass WebexObject for all model classes
* Removed unneeded imports

dtibbe commented:
I've fixed some code errors and introduced a common WebexObject super class for all Webex API objects. That one should encapsulate common behavior such as the ToString().

dtibbe committed:
* Updated readme file
